### PR TITLE
Jc/mini cleanup

### PR DESCRIPTION
**Pull request by juchom**
# Pull Request

## What does this PR do?
This is mostly a cosmetic cleanup.

The `Constants.cs` is now marked as internal. There is no need to surface this to the end-user. We don't want them to use it their own project. We want to be able to change what is in this file without creating breaking changes in the future.

The `Index.cs` was pretty huge, I've splitted it in smaller files that are big sections of the definition of an index and document management. This will allow new comers to study the code easily and make refactoring, upgrades, maintenance easier for developers.

image

Let me know if you like it.

### CODE REVIEW

File: src/Meilisearch/Meilisearch.csproj
Comment (brunoocasali) [inline src/Meilisearch/Meilisearch.csproj:22]: Just a curiosity, why do you need to make this change as well? The `partial` keyword doesn't work out of the box?

**Review by brunoocasali — DISMISSED** [review]
Thanks a lot for your contribution @juchom you rocks! 👨‍🎤 

Meilisearch is better with contributors like you!

PS: I'll let @alallema review as well since she is the new official maintainer of this package :)

File: src/Meilisearch/Meilisearch.csproj
Comment (juchom) [reply src/Meilisearch/Meilisearch.csproj:22]: >Just a curiosity, why do you need to make this change as well?

The csproj modification is for the nesting like this

image

Without the csproj modification it would looks like this
image

>The `partial` keyword doesn't work out of the box?

The `partial` keyword is mandatory if you want to split a class across several files.

Comment (brunoocasali) [reply src/Meilisearch/Meilisearch.csproj:22]: Oh nice, good to know!

**Review by alallema — COMMENTED** [review]
Hi @juchom,
Thanks a lot for this PR is really great I love it. I just have a hesitation about the settings, I wonder if the following files (`Index.Attributes.cs`, `Index.RankingsRules.cs`, `Index.StopWords.cs` and `Index.Synonyms.cs`) should be together in the `Index.Settings.cs` because there are all related to the Settings what do you think?

**Comment by juchom** [comment]
Hi @alallema,

Good question, I had the exact same one when I authored this PR.

The main issue I see with having all the Settings in one file is there is a lot of Settings
DisplayableAttributes
DistinctAttributes
FilterableAttributes
SearchAttribultes
SortableAttributes
All the attributes above (198 lines)

RankingRules (47 lines)

StopWords (47 lines)

Synonyms (48 lines)

While writing this reply, I think we could merge all of this in one file as you suggest. The settings part of the client is not moving that much.

I think it would be best to merge everything in a Index.Settings.cs file. Let me know if you prefer this solution.

**Comment by alallema** [comment]
> The main issue I see with having all the Settings in one file is there are a lot of Settings
> The settings part of the client is not moving that much.

I agree with you, and indeed this part would not change but others will come so maybe it is better to leave as you did.
My bad I could have approved directly 😊

**Review by alallema — DISMISSED** [review]
LGTM! 🔥 🔥 🔥 
Thanks again for this PT @juchom

**Comment by alallema** [comment]
@juchom, did you prefer to handle the conflict or I can to do it myself?

**Comment by juchom** [comment]
> @juchom, did you prefer to handle the conflict or I can to do it myself?

Let me handle this !

**Review by alallema — DISMISSED** [review]
LGTM! Perfect 🎉

**Review by alallema — APPROVED** [review]

**Comment by alallema** [comment]
bors merge

**Pull request merged**
